lib/Catalyst/Plugin/PrometheusTiny.pm

```diff
@@ -117,35 +126,37 @@ after finalize => sub {
 
 before setup_components => sub {
     my $class = shift;
+
+    # initialise prometheus instance pre-fork and setup lexicals
+    $class->prometheus;
+
     return
       if $class->config->{'Plugin::PrometheusTiny'}{no_default_controller};
 
+    eval qq(
+        package Catalyst::Plugin::PrometheusTiny::Controller;
+        use base 'Catalyst::Controller';
+
+        sub begin : Private { }
+        sub end : Private   { }
+
+        sub metrics : Path($metrics_endpoint) Args(0) {
+            my ( \$self, \$c ) = \@_;
+            my \$res = \$c->res;
+            \$res->content_type("text/plain");
+            \$res->output( \$c->prometheus->format );
+        }
+        1;
+
+    );
```

Comment:
I know it's probably silly but ...

1. You don't check if the `eval` succeeded. Long-term, if this code grows, it will be problematic.
2. I would prefer a different delimiter for `qq` because I see it's used _inside_ the code block and I was curious why that wasn't a syntax error.